phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/behavior/ListWidgetSelectorBehavior.java

```diff
@@ -0,0 +1,85 @@
+package org.phoenicis.javafx.components.behavior;
+
+import javafx.event.ActionEvent;
+import javafx.scene.control.ToggleButton;
+import org.phoenicis.javafx.components.control.ListWidgetSelector;
+import org.phoenicis.javafx.components.skin.ListWidgetSelectorSkin;
+import org.phoenicis.javafx.views.common.widgets.lists.ListWidgetType;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * The behavior for the {@link ListWidgetSelector} component
+ */
+public class ListWidgetSelectorBehavior
+        extends BehaviorBase<ListWidgetSelector, ListWidgetSelectorSkin, ListWidgetSelectorBehavior> {
+    /**
+     * Constructor
+     *
+     * @param control The control associated with this behavior
+     * @param skin The skin associated with this behavior
+     */
+    public ListWidgetSelectorBehavior(ListWidgetSelector control, ListWidgetSelectorSkin skin) {
+        super(control, skin);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialise() {
+        // set the behavior for the icons list toggle button
+        getSkin().getIconsListButton().addEventFilter(ActionEvent.ANY, this::eventFilter);
+        getSkin().getIconsListButton()
+                .setOnAction(event -> {
+                    getControl().setSelected(ListWidgetType.ICONS_LIST);
+                    getOnSelect().ifPresent(consumer -> consumer.accept(ListWidgetType.ICONS_LIST));
+                });
+
+        // set the behavior for the compact list toggle button
+        getSkin().getCompactListButton().addEventFilter(ActionEvent.ANY, this::eventFilter);
+        getSkin().getCompactListButton()
+                .setOnAction(event -> {
+                    getControl().setSelected(ListWidgetType.COMPACT_LIST);
+                    getOnSelect().ifPresent(consumer -> consumer.accept(ListWidgetType.COMPACT_LIST));
+                });
+
+        // set the behavior for the details list toggle button
+        getSkin().getDetailsListButton().addEventFilter(ActionEvent.ANY, this::eventFilter);
+        getSkin().getDetailsListButton()
+                .setOnAction(event -> {
+                    getControl().setSelected(ListWidgetType.DETAILS_LIST);
+                    getOnSelect().ifPresent(consumer -> consumer.accept(ListWidgetType.DETAILS_LIST));
+                });
+
+        // ensure that if at startup a selected type is set select it
+        Optional.ofNullable(getControl().getSelected())
+                .ifPresent(selected -> getSkin().getListButton(selected).fire());
+
+        // change the selection based on a change in the selected property
+        getControl().selectedProperty()
+                .addListener(invalidatedEvent -> getSkin().getListButton(getControl().getSelected()).fire());
+    }
+
+    /**
+     * An event filter to prevent the unselectation of all buttons
```

Comment:
selection?

Reply:
deselection seems to be the better English word instead of unselection.

Reply:
Was just about the "selectation" part because that just doesn't seem to exist.